kubernetes/classy-vision.yaml

```diff
@@ -0,0 +1,112 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: classy-vision-worker-1
+  labels:
+    job-name: classy-vision
+    worker: "1"
+spec:
+  restartPolicy: "OnFailure"
+  containers:
+  - name: classy-vision
+    image: torchelastic/examples:0.1.0rc1
+    args:
+    - "s3://torchelastic-shjiaxin-1h71m-s3bucket-m1b9b9pjldqw/petctl/shjiaxin/my_job/main.py"
+    - "--config_file"
+    # Get configuration file from here
+    # https://github.com/facebookresearch/ClassyVision/blob/e7e02b0e4d7fa4cfb8bc909f0e6bb9488a1073b8/classy_vision/configs/resnet50_synthetic_image_classy_config.json
+    - "/data/classy_vision/resnet50_synthetic_image_classy_config.json"
+    - "--checkpoint_folder"
+    - "/data/classy_vision/checkpoint"
+    env:
+    - name: RDZV_ENDPOINT
+      value: "etcd:2379"
+    - name: JOB_ID
+      value: classy-vision-2
+    - name: MIN_SIZE
+      value: "1"
+    - name: MAX_SIZE
+      value: "3"
+    - name: SIZE
+      value: "2"
+    volumeMounts:
+    - name: persistent-storage
+      mountPath: /data
+    - name: dshm
+      mountPath: /dev/shm
+    name: dshm
+  volumes:
+  - name: persistent-storage
+    persistentVolumeClaim:
+      claimName: efs-claim
+  - name: dshm
+    emptyDir:
+      medium: Memory
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: classy-vision-worker-1
+spec:
+  selector:
+    job-name: classy-vision
+    worker: "1"
+  clusterIP: None
+
+---
+apiVersion: v1
+kind: Pod
```

Comment:
Any reason why you did not use `kind: Deployment` here?

Reply:
There're two reason we can not use Deployment.

1. It underneath uses `ReplicaSet` which is good for long running services. Even if the task finishes, it will try to bring up new tasks to make sure desired count is same as what user specified.  `v1/Job` is a better choice. 
2. If I understand elastic project correctly, worker talks with each other.  In this case,  we can not create service for each individual pods because they share exact same labels. If we create service and match the label, it could be multiple endpoints and behavior like a Load balancer. Instead, we need to create identical labels for each pod, and then create [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) to make sure pod can talk to other pods using hostname.

Reply:
Hello @Jeffwan, creating a K8s `Job` actually sounds like a great idea. Would it be too hard to make this change in your PR?

Reply:
Job won't work here because there're no way to discover peer using pod hostname. 

Pod is low level primitive and user will have more flexibility to control. If we consider high level primitive, `Statefulset` will be a better option. Consider this is just a example, once operator is ready, no one will use this way, I will try to change to statefulset.

Reply:
Some updates on it. Currently, statefulset can create index job and make sure each task can talk with other tasks using hostname (By default, hostname is not resolvable, It need some dns setting on pod template).  But stateful doesn't support `OnFailure` or `Never` restartPolicy (https://github.com/kubernetes/community/issues/458) which means each task is complete, APIServer will bring it up again because statefulset is designed for long running jobs. Because of this, for naive example, I think pod + headless service is good enough. Operator will provide better abstraction for users later.

Reply:
Thanks and agreed. Let's keep it this way since it appears that it'll be very hard to change this to something other than a headless service.